File: 1813-maximum-erasure-value/maximum-erasure-value.cpp
class Solution {
public:
    int maximumUniqueSubarray(vector<int>& nums) {
        unordered_map<int, int> map;
        int score = 0, sum = 0, i = 0, j = 0, n = nums.size();

        while (j < n) {
            map[nums[j]]++;
            sum += nums[j];
            while (map[nums[j]] > 1) {
                map[nums[i]]--;
                sum -= nums[i];
                i++;
            }
            j++;

            score = max(sum, score);
        }

        return score;
    }
};

File: 1813-maximum-erasure-value/maximum-erasure-value.cs
public class Solution {
    public int MaximumUniqueSubarray(int[] nums) {
        var dict = new Dictionary<int, int>();
        var score = 0;
        var sum = 0;

        int i=0, j=0;
        while (j < nums.Length) {
            sum += nums[j];
            if (dict.ContainsKey(nums[j])) dict[nums[j]]++;
            else dict[nums[j]] = 1;

            while (dict[nums[j]] > 1) {
                dict[nums[i]]--;
                sum -= nums[i];
                i++;
            }
            j++;

            score = Math.Max(score, sum);
        }

        return score;
    }
}